yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -63,7 +65,21 @@ var assessMigrationCmd = &cobra.Command{
 		validateSourceSchema()
 		validatePortRange()
 		validateSSLMode()
-		validateAssessmentMetadataDirFlag()
+		if cmd.Flags().Changed("assessment-metadata-dir") {
+			validateAssessmentMetadataDirFlag()
+			cmd.Flags().VisitAll(func(f *pflag.Flag) {
+				if f.Changed {
+					if strings.Contains(strings.ToLower(f.Name), "source") && f.Name != "source-db-type" {
+						utils.ErrExit("No need of `--source-*` connection related flags as `--assessment-metadata-dir` is provided.\nPlease re-run the command without these flags")
+					}
+				}
+			})
+		} else {
+			cmd.MarkFlagRequired("source-db-user")
```

Comment:
We are essentially marking the flag as required at runtime, not compile time, here, essentially, right? 
How does that work? Does this throw a validation error here then?

Reply:
Yes, it throws this error 
```
Error: required flag(s) "source-db-schema" not set
```

Reply:
> We are essentially marking the flag as required at runtime, not compile time, here, essentially, right?

Yes